tests/Silverback.Integration.Tests.E2E/Kafka/MessageValidationTests.cs

```diff
@@ -0,0 +1,344 @@
+// Copyright (c) 2020 Sergio Aquilini
+// This code is licensed under MIT license (see LICENSE file for details)
+
+using System;
+using System.Collections.Generic;
+using System.ComponentModel.DataAnnotations;
+using System.Text;
+using System.Threading.Tasks;
+using FluentAssertions;
+using Microsoft.Extensions.DependencyInjection;
+using Silverback.Messaging.Configuration;
+using Silverback.Messaging.Messages;
+using Silverback.Messaging.Publishing;
+using Silverback.Messaging.Validation;
+using Silverback.Tests.Integration.E2E.TestHost;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace Silverback.Tests.Integration.E2E.Kafka
+{
+    public class MessageValidationTests : KafkaTestFixture
```

Comment:
I think that we should move the theory with all cases to a unit test, that tests the behavior in isolation, with mocks.
Beside the better performance, that would also allow you to check the exceptions content and that the warnings are actually being written.

Here, as end-to-end test, we can keep just a single case, only to ensure that the pipeline works. (I mean a single case per constellation of setting and valid/invalid message.)